httpx/config.py

```diff
@@ -8,9 +8,18 @@
 from .__version__ import __version__
 from .utils import get_ca_bundle_from_env, get_logger
 
+
+class UnsetType:
```

Comment:
Unfortunately can't use this pattern

```py
UnsetType = NewType('UnsetType', object)
UNSET = UnsetType(object())
```

because `NewType` does not allow `isinstance()` on it, and it was needed otherwise mypy was complaining in `TimeoutConfig`.

Reply:
Yup, well done here.

From Python 3.8, `typing.Literal` would allow to solve this quite elegantly:

```python
import typing

UNSET = object()
MyCustomType = typing.Union[bool, typing.Literal[UNSET]]
```

Not sure that there'll be a backport to 3.6 and 3.7 unfortunately, but just thought it was interesting to share. :)

Reply:
It's available in the `typing_extensions` package https://github.com/python/typing/tree/master/typing_extensions

But I guess it's not worth adding a new dependency just for this (since we have the class workaround anyway).

Reply:
@florimondmanca I think `typing.Literal` is not valid in this usage. `UNSET` is not a `Literal`.